tests/test_find_command_line_interface.py

```diff
@@ -64,24 +62,11 @@
 ]
 
 
-@contextmanager
-def redirect_stderr(new_target=None):
-    "Temporarily redirect all output to stderr to new_target."
-    if new_target is None:
-        new_target = StringIO()
-    old_target = sys.stderr
-    try:
-        sys.stderr = new_target
-        yield
-    finally:
-        sys.stderr = old_target
-
-
 class TestFindCommandLineInterface:
     @staticmethod
     def _parse(args):
-        with open(os.devnull, "w") as null:
-            return parse_filter_arg(args, file=null)
+        with redirect_stderr(StringIO()):
```

Comment:
This is more of a question for understanding- I was looking at this function, and it seems that it has created errors to people before (https://stackoverflow.com/questions/38291744/redirect-stderr-does-not-work-python-3-5). I was wondering if this is resolved because it is a @staticmethod or whether we should accommodate for this.

Reply:
That StackOverflow question is misusing `redirect_stderr`. See the answer below: "You need to actually write to stderr, it is not a tool to catch exceptions."

The use of a `@staticmethod` here is to indicate that the method implementation is not dependent on the current instance `self` (as in a "normal" method) or the class `cls` (as in a `@classmethod`). A `@staticmethod` is equivalent to a free function (defined outside of a class) but has the benefit of being "scoped" through its definition in the class. To be honest, I rarely use static methods in Python because they are kind of awkward. Some developers argue that you should always use a free function (putting `def _parse` outside the class) but I didn't want to change this part of the test. See also: https://stackoverflow.com/questions/136097/difference-between-staticmethod-and-classmethod